pkg/controllers/knative/deployer_controller.go

```diff
@@ -414,6 +418,25 @@ func (r *DeployerReconciler) constructLabelsForDeployer(deployer *knativev1alpha
 	return labels
 }
 
+func (r *DeployerReconciler) constructAnnotationsForDeployer(deployer *knativev1alpha1.Deployer) map[string]string {
+	// make a little extra space just in case
+	annotations := make(map[string]string, len(deployer.ObjectMeta.Annotations)+2)
+
+	// pass through existing annotations
+	for k, v := range deployer.ObjectMeta.Annotations {
+		annotations[k] = v
+	}
+
+	if deployer.Spec.Scale.Min != nil {
+		annotations["autoscaling.knative.dev/minScale"] = strconv.Itoa(int(*deployer.Spec.Scale.Min))
```

Comment:
consider something like this to avoid type downcasting, or switch the type in the resource from int32 to int

```suggestion
		annotations["autoscaling.knative.dev/minScale"] = fmt.Sprintf("%d", *deployer.Spec.Scale.Min)
```

Reply:
Change made! Is the idea that casting is a heavy activity and something to be avoided where possible?